src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -708,6 +769,76 @@ private String simpleStudyName(String scopedName){
         return scopedName.replaceFirst(" \\[.*\\]", "");
     }
 
+    private void validateObservationValue(Trait variable, String value,
+                                          String columnHeader, ValidationErrors validationErrors, int row) {
+        if(StringUtils.isBlank(value)) {
+            log.debug(String.format("skipping validation of observation because there is no value.\n\tvariable: %s\n\trow: %d", variable.getObservationVariableName(), row));
+            return;
+        }
+
+        switch(variable.getScale().getDataType()) {
+            case NUMERICAL:
+                Optional<BigDecimal> number = validNumericValue(value);
+                if (number.isEmpty()) {
+                    addRowError(columnHeader, "Non-numeric text detected detected", validationErrors, row);
+                }
+                else if (!validNumericRange(number.get(), variable.getScale())) {
+                    addRowError(columnHeader, "Value outside of min/max range detected", validationErrors, row);
+                }
+                break;
+            case DATE:
+                if (!validDateValue(value)) {
+                    addRowError(columnHeader, "Incorrect date format (YYYY-MM-DD) detected", validationErrors, row);
+                }
+                break;
+            case ORDINAL:
+                if (!validCategory(variable.getScale().getCategories(), value)) {
+                    addRowError(columnHeader, "Undefined ordinal category detected", validationErrors, row);
+                }
+                break;
+            case NOMINAL:
+                if (!validCategory(variable.getScale().getCategories(), value)) {
+                    addRowError(columnHeader, "Undefined nominal category detected", validationErrors, row);
+                }
+                break;
+            default:
+                break;
+        }
+
+    }
+    private Optional<BigDecimal> validNumericValue(String value) {
+        BigDecimal number;
+        try {
+            number = new BigDecimal(value);
+        } catch (NumberFormatException e) {
+            return Optional.empty();
+        }
+        return Optional.of(number);
+    }
+
+    private boolean validNumericRange(BigDecimal value, Scale validValues) {
+        if (value.compareTo(BigDecimal.valueOf(validValues.getValidValueMin())) >= 0 &&
```

Comment:
This code will throw a `NullPointerException` if `validValues.getValidValueMin` or `validValues.getValidValueMax` are `null`

Reply:
Thanks for spotting that, pushed changes.